### PR TITLE
Code changes for unit-testing

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------
This PR adds some code changes to the FreeRTOS_DNS files as a precursor to unit-testing. This would be helpful while reviewing the PR for unit-testing https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/221.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: include/FreeRTOS_DNS.h [hunk 1]
```diff
@@ -36,6 +36,76 @@
     #include "IPTraceMacroDefaults.h"
 
 
+    #if ( ipconfigBYTE_ORDER == pdFREERTOS_LITTLE_ENDIAN )
```
Comment (alfred2g) [inline include/FreeRTOS_DNS.h:39]: are these defined values that were moved form the source file to the header file going to be used by other files, or it is just a change to accommodate unit testing ?

**Review by alfred2g — APPROVED** [review]

File: FreeRTOS_DNS.c [hunk 2]
```diff
@@ -431,7 +266,7 @@
         }
 
 /** @brief The list of all callback structures. */
-        static List_t xCallbackList;
+        _static List_t xCallbackList;
```
Comment (aggarg) [inline FreeRTOS_DNS.c:269]: Why this change?

Comment (AniruddhaKanhere) [reply FreeRTOS_DNS.c:269]: _static is #defined as static. But for unit-testing and cbmc checks, _static is defined away to nothing. This helps in covering all branches of the source code and generating scenarios that are not interdependent.
Such as, some test cases need the `xCallbackList` to be `NULL`. But once a function that initializes the list is called, we cannot test the NULL conditions later on...

**Review by aggarg — APPROVED** [review]
